api/src/main/java/com/epam/pipeline/manager/metadata/MetadataManager.java

```diff
@@ -343,6 +355,20 @@ public Set<String> getMetadataKeys(final AclClass entityClass) {
         return keys;
     }
 
+    @SuppressWarnings("PMD.AvoidCatchingGenericException")
+    public Map<String, String> prepareCustomInstanceTags(final PipelineRun run) {
+        try {
+            if (!CloudProvider.AWS.equals(run.getInstance().getCloudProvider())) {
+                return Collections.emptyMap();
+            }
+            final Map<String, String> customTags = resolveCommonCustomInstanceTags(run);
+            return resolveInstanceTagsFromMetadata(run.getDockerImage(), customTags);
```

Comment:
I would suggest using a simpler approach for collecting tags from various sources.

```suggestion
            final Map<String, String> implicitTags = resolveCommonCustomInstanceTags(run);
            final Map<String, String> explicitTags = resolveInstanceTagsFromMetadata(run);
            return CommonUtils.mergeMaps(implicitTags, explicitTags);
```